lib/spack/llnl/util/filesystem.py

```diff
@@ -1411,7 +1411,7 @@ def search_paths_for_executables(*path_hints):
         if not os.path.isdir(path):
             continue
 
-        executable_paths.append(path)
+        executable_paths.append(os.path.abspath(path))
```

Comment:
Now `../gcc/bin` will work, but `../gcc/` won't. You should use:
```python
path = os.path.abspath(path)
```

Reply:
Ok, its fixed now.